src/QMCDrivers/WFOpt/EngineHandle.h

```diff
@@ -143,9 +151,22 @@ class LMYEngineHandle : public EngineHandle
           le_der_samp[0] * static_cast<FullPrecValue>(dlogpsi_array.getValue(j, ib));
     }
 
+    int ip = omp_get_thread_num();
```

Comment:
There looks to be a potential disaster data race above this on der_rat_samp and le_der_samp.

The only id you should care about in a threaded section is the crowd id, it may or may not have any relation to the omp thead id. Crowds may be called at the same time on any collection of thread_id's the concurrency context you're in is identified by the crowd id and you should avoid needing to know it.